apps/webpack.config.js

```diff
@@ -75,6 +84,43 @@ const nodePolyfillConfig = {
       process: 'process/browser',
       timers: 'timers-browserify',
     }),
+    new CircularDependencyPlugin({
+      // exclude detection of files based on a RegExp
+      exclude: /node_modules|build/,
+      // add errors to webpack instead of warnings
+      failOnError: true,
+      // allow import cycles that include an asyncronous import,
+      // e.g. via import(/* webpackMode: "weak" */ './file.js')
+      allowAsyncCycles: false,
+      // set the current working directory for displaying module paths
+      cwd: process.cwd(),
+      onStart: () => {
+        seenCircles.clear();
+        seenCircles = new Set(Array.from(circleSet));
+      },
+      onDetected: ({module: webpackModuleRecord, paths, compilation}) => {
+        const pathString = paths.join(' -> ');
+        if (!circleSet.has(pathString)) {
+          compilation.errors.push(
+            new Error(
+              `Circular Dependency Checker : New Circular Dependency found : ${pathString}`
+            )
+          );
+        }
+        seenCircles.delete(pathString);
+      },
+      onEnd: ({compilation}) => {
+        if (seenCircles.size > 0) {
+          compilation.errors.push(
```

Comment:
What about adding this note to `compilation.warnings` rather than `compilation.errors`? Seems like this should operate at a lower severity level than new dependencies being found